### PR TITLE
adds linked-list exercise

### LINKED ISSUE #117: Implement linked-list
PR #109

### PR DESCRIPTION
**Pull request by Atomic-Germ**
modified:   config.json
	new file:   exercises/practice/linked-list/.docs/instructions.md
	new file:   exercises/practice/linked-list/.meta/config.json
	new file:   exercises/practice/linked-list/.meta/proof.ci.wren
	new file:   exercises/practice/linked-list/.meta/tests.toml
	new file:   exercises/practice/linked-list/LICENSE
	new file:   exercises/practice/linked-list/linked-list.spec.wren
	new file:   exercises/practice/linked-list/linked-list.wren
	new file:   exercises/practice/linked-list/package.wren

closes #117

### CODE REVIEW

File: exercises/practice/linked-list/.meta/config.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "blurb": "Implement a doubly linked list.",
+  "authors": [
+    "Atomic-Germ"
+  ],
+  "contributors": [],
```
Comment (glennj) [inline exercises/practice/linked-list/.meta/config.json:6]: Suggestion: run `bin/configlet fmt -u -e $slug`

File: exercises/practice/linked-list/linked-list.spec.wren [hunk 2]
```diff
@@ -0,0 +1,198 @@
+import "wren-testie/testie" for Testie, Expect
+import "./linked-list" for LinkedList
+
+Testie.test("Linked List") { |do, skip|
+
+  do.test("poop gets last element from the list") {
```
Comment (glennj) [inline exercises/practice/linked-list/linked-list.spec.wren:6]: 8-year-old me says "He said 'poop'!"

File: exercises/practice/linked-list/.meta/proof.ci.wren [hunk 3]
```diff
@@ -0,0 +1,41 @@
+class LinkedList {
+  construct new() {
+    _list = []
+  }
```
Comment (glennj) [inline exercises/practice/linked-list/.meta/proof.ci.wren:4]: Using a list object to hold the data makes this exercise trivial.

I'd prefer if we did the same as javascript: tell students _not_ to use a list to solve it: https://github.com/exercism/javascript/blob/main/exercises/practice/linked-list/.docs/instructions.append.md

**Comment by glennj** [comment]
Since we're both creating exercises, I'd suggest we use issues to track exercises we claim. I did this on the awk track with success: create an issue to indicate you have claimed an issue and assign it to yourself. Put `Closed #123` (the number of the issue) in the PR description to link them (the issue gets auto-closed when the PR is merged).

File: exercises/practice/linked-list/linked-list.spec.wren (hunk 2 above)
Comment (Atomic-Germ) [reply exercises/practice/linked-list/linked-list.spec.wren:6]: 🤣 Whoops!

File: exercises/practice/linked-list/.meta/config.json (hunk 1 above)
Comment (Atomic-Germ) [reply exercises/practice/linked-list/.meta/config.json]: I definitely thought I ran that 🧐

File: exercises/practice/linked-list/.meta/proof.ci.wren (hunk 3 above)
Comment (Atomic-Germ) [reply exercises/practice/linked-list/.meta/proof.ci.wren:4]: Fair enough

File: exercises/practice/linked-list/.docs/instructions.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+* `shift` (*remove value at front*).
+* `unshift` (*insert value at front*);
+
+To keep your implementation simple, the tests will not cover error
+conditions. Specifically: `pop` or `shift` will never be called on an
+empty list.
+
+If you want to know more about linked lists, check [Wikipedia](https://en.wikipedia.org/wiki/Linked_list).
+
+NOTE: Do not use a library to implement this exercise. Do not use a backing array to implement this exercise.
```
Comment (glennj) [inline exercises/practice/linked-list/.docs/instructions.md:30]: This file is essentially copied from the problem-specifications repo. We should not edit it.

Add track-specific stuff to the `instructions.append.md` file.

File: exercises/practice/linked-list/.meta/proof.ci.wren (hunk 3 above)
Comment (joshgoebel) [reply exercises/practice/linked-list/.meta/proof.ci.wren:4]: Yes, the proofs  - **while not needing to be perfect or fully optimized** - should at least be a "reasonable good faith" attempt to solve the problem as specified - which in this case requires writing an actual linked list.  Also they should preferably use Wren-isms and consistent Wren-like naming when they can. 

@Atomic-Germ Thanks for keeping them coming though! :-)  Awesome to see the Wren track growing so fast lately.

**Comment by glennj** [comment]
@Atomic-Germ were you going to continue? I'd be happy to help out.

**Comment by Atomic-Germ** [comment]
@glennj help would be welcome, thank you

**Comment by glennj** [comment]
@Atomic-Germ I took the liberty of cloning your repo and committing to it.

File: exercises/practice/linked-list/.docs/instructions.append.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+# note
```
Comment (joshgoebel) [inline exercises/practice/linked-list/.docs/instructions.append.md:1]:
```suggestion
# Notes
```

File: exercises/practice/linked-list/linked-list.spec.wren [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+
+Testie.test("Linked List") { |do, skip|
+
+  do.test("pop gets last element from the list") {
+    var list = LinkedList.new()
+    list.push(7)
+    Expect.value(list.pop()).toEqual(7)
+  }
+
+  do.test("push/pop add and remove at the end of the list respectively") {
```
Comment (joshgoebel) [inline exercises/practice/linked-list/linked-list.spec.wren:12]: Shouldn't all of these be `skip.test` other than the first?

**Review by joshgoebel — CHANGES_REQUESTED** [review]
Looks good except the 2 things I mentioned.

File: exercises/practice/linked-list/.docs/instructions.append.md (hunk 5 above)
Comment (glennj) [reply exercises/practice/linked-list/.docs/instructions.append.md:1]: The level-1 header of the append file is discarded by the process that joins the files. It doesn't matter what goes in there.

Comment (joshgoebel) [reply exercises/practice/linked-list/.docs/instructions.append.md:1]: Good to know, but no reason to be sloppy IMHO. :-)

File: exercises/practice/linked-list/linked-list.spec.wren (hunk 6 above)
Comment (glennj) [reply exercises/practice/linked-list/linked-list.spec.wren]: Done

**Review by joshgoebel — APPROVED** [review]

**Comment by Atomic-Germ** [comment]
> @Atomic-Germ I took the liberty of cloning your repo and committing to it.

Thank you!

**Comment by joshgoebel** [comment]
@Atomic-Germ Thanks so much for helping and getting the ball rolling on this one!

**Pull request merged**